WordPress/src/main/java/org/wordpress/android/datasets/PeopleTable.java

```diff
@@ -48,122 +71,214 @@ public static void reset(SQLiteDatabase db) {
+        savePeople(FOLLOWERS_TABLE, peopleList, localTableBlogId, isFreshList);
     }
 
-    public static void savePeople(List<Person> peopleList, int localTableBlogId, boolean isFreshList) {
+    public static void saveEmailFollowers(List<Person> peopleList, int localTableBlogId, boolean isFreshList) {
+        savePeople(EMAIL_FOLLOWERS_TABLE, peopleList, localTableBlogId, isFreshList);
+    }
+
+    private static void savePeople(String table, List<Person> peopleList, int localTableBlogId, boolean isFreshList) {
         getWritableDb().beginTransaction();
         try {
-            //We have a fresh list, remove the previous list of people in case it was deleted on remote
+            // We have a fresh list, remove the previous list of email followers in case it was deleted on remote
             if (isFreshList) {
-                PeopleTable.deletePeopleForLocalBlogId(localTableBlogId);
+                PeopleTable.deletePeople(table, localTableBlogId);
             }
 
             for (Person person : peopleList) {
-                PeopleTable.save(person);
+                PeopleTable.save(table, person, getWritableDb());
             }
             getWritableDb().setTransactionSuccessful();
         } finally {
             getWritableDb().endTransaction();
         }
     }
 
-    public static int getPeopleCountForLocalBlogId(int localTableBlogId) {
-        String[] args = new String[]{Integer.toString(localTableBlogId)};
-        String sql = "SELECT COUNT(*) FROM " + PEOPLE_TABLE + " WHERE local_blog_id=?";
-        return SqlUtils.intForQuery(getReadableDb(), sql, args);
+    public static void deletePeopleForLocalBlogId(int localTableBlogId) {
+        deletePeople(TEAM_TABLE, localTableBlogId);
+        deletePeople(FOLLOWERS_TABLE, localTableBlogId);
+        deletePeople(EMAIL_FOLLOWERS_TABLE, localTableBlogId);
     }
 
-    public static void deletePeopleForLocalBlogId(int localTableBlogId) {
+    private static void deletePeople(String table, int localTableBlogId) {
         String[] args = new String[]{Integer.toString(localTableBlogId)};
-        getWritableDb().delete(PEOPLE_TABLE, "local_blog_id=?", args);
-    }
-
-    public static void deletePeopleForLocalBlogIdExceptForFirstPage(int localTableBlogId) {
-        int size = getPeopleCountForLocalBlogId(localTableBlogId);
-        int fetchLimit = PeopleUtils.FETCH_USERS_LIMIT;
-        if (size > fetchLimit) {
-            int deleteCount = size - fetchLimit;
-            String[] args = new String[] {
-                    Integer.toString(localTableBlogId),
-                    Integer.toString(deleteCount)
-            };
-            getWritableDb().delete(PEOPLE_TABLE, "local_blog_id=?1 AND person_id " +
-                    "IN (SELECT person_id FROM " + PEOPLE_TABLE + " WHERE local_blog_id=?1 " +
-                    "ORDER BY display_name DESC LIMIT ?2)", args);
+        getWritableDb().delete(table, "local_blog_id=?1", args);
```

Comment:
TIL - didn't know about the `?1` notation, but guess it's not necessary here with only one argument. No need to take action though. 👍